### PR TITLE
feat: enable kalium to pick a remote API version for each user/auth session independently

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

AR now support only one remote API version

### Solutions

use the new kalium `AutoVersionAuthScopeUseCase` to provide versioned auth scope that can be used for login/register

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1053/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1053/4/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/home/conversations/details/participants/GroupConversationParticipantItem.kt [hunk 1]
```diff
@@ -41,6 +44,19 @@ fun GroupConversationParticipantItem(
                     searchQuery = searchQuery,
                     modifier = Modifier.weight(weight = 1f, fill = false)
                 )
+                if (uiParticipant.isSelf) {
```
Comment (typfel) [inline app/src/main/kotlin/com/wire/android/ui/home/conversations/details/participants/GroupConversationParticipantItem.kt]: Should not be included?

File: app/src/main/kotlin/com/wire/android/util/lifecycle/ConnectionPolicyManager.kt [hunk 2]
```diff
@@ -122,7 +122,7 @@ class ConnectionPolicyManager @Inject constructor(
         } else {
             ConnectionPolicy.DISCONNECT_AFTER_PENDING_EVENTS
         }
-        coreLogic.getSessionScope(userId).setConnectionPolicy(connectionPolicy)
+//        coreLogic.getSessionScope(userId).setConnectionPolicy(connectionPolicy)
```
Comment (typfel) [inline app/src/main/kotlin/com/wire/android/util/lifecycle/ConnectionPolicyManager.kt]: Undo?

File: app/src/main/kotlin/com/wire/android/ui/authentication/login/email/LoginEmailViewModel.kt [hunk 3]
```diff
@@ -25,10 +23,10 @@ import javax.inject.Inject
 @ExperimentalMaterialApi
 @HiltViewModel
 class LoginEmailViewModel @Inject constructor(
-    private val loginUseCase: LoginUseCase,
+//    private val loginUseCase: LoginUseCase,
```
Comment (typfel) [inline app/src/main/kotlin/com/wire/android/ui/authentication/login/email/LoginEmailViewModel.kt]: Delete

**Review by typfel — CHANGES_REQUESTED** [review]

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1053/5/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-1053/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-1053/9/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4550122)(4550122)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4550122)(4550122)-dev-release.apk)

**Review by typfel — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-1053/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4555703)(4555703)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4555703)(4555703)-dev-release.apk)

**Pull request merged**